### ISSUE TITLE
CodeAnalysisTreatWarningsAsErrors unreliable

### ISSUE THREAD

**Issue by SeSamSeg** [dotnet/roslyn-analyzers#7829]
## To reproduce

Have:

csproj
``` 
  <PropertyGroup>
    <!-- Compiler and analysis erros on build -->
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    <!-- Don't error on analysis warnings, this makes it possible to have more layers in the static analysis, otherwise it's only error or suggestion -->
    <CodeAnalysisTreatWarningsAsErrors>false</CodeAnalysisTreatWarningsAsErrors>
  </PropertyGroup>
```

.editorconfig
```
dotnet_diagnostic.CA2007.severity = error
```

## Current behavior

```
C:\CA2007_build_twice_bug> dotnet build .\CA2007_build_twice_bug.csproj
Restore complete (0,2s)
  CA2007_build_twice_bug net8.0 succeeded with 1 warning(s) (0,2s) → bin\Debug\net8.0\CA2007_build_twice_bug.dll
    C:\Users\sam\Sources\CA2007_build_twice_bug\Class1.cs(7,15): warning CA2007: Consider calling ConfigureAwait on the awaited task (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2007)

Build succeeded with 1 warning(s) in 0,9s
C:\CA2007_build_twice_bug> dotnet build .\CA2007_build_twice_bug.csproj
Restore complete (0,2s)
  CA2007_build_twice_bug net8.0 succeeded (0,1s) → bin\Debug\net8.0\CA2007_build_twice_bug.dll

Build succeeded in 0,8s
```

## Expected behavior

That the combination of TreatWarningsAsErrors==true and CodeAnalysisTreatWarningsAsErrors==false would treat error as error.

## Rational

I want to have msbuild, nuget and compiler warnings as errors, but don't loose the option to mark some analysis as warnings to still have the layer in rules and not have only Errors and Messages. That's the purpose of CodeAnalysisTreatWarningsAsErrors. But it seems that CodeAnalysisTreatWarningsAsErrors does not use the .editorconfig for restoring errors to warnings and only the rules defined by AnalysisMode and AnalysisLevel.

## Example

[CA2007_build_twice_bug.zip](https://github.com/user-attachments/files/28016964/CA2007_build_twice_bug.zip)

## Notes

- A quick search on CodeAnalysisTreatWarningsAsErrors seems to indicate a lot of issues are related to this and probably also a lot of issues not even mentioning CodeAnalysisTreatWarningsAsErrors as well (like VS gives other errors than dotnet)
- Tested:
  - 10.0.300
  - 11.0.100-preview.4.26230.115

**Comment by Frulfump** [comment]
Do you get the warning again if you do `dotnet build --no-incremental .\CA2007_build_twice_bug.csproj`?
With that said see below (might be a better fit for the dotnet SDK repo even before the consolidation below)

From the project README.md

>Repository Consolidation
>
>The Microsoft.CodeAnalysis.NetAnalyzers package moved into the [dotnet/sdk](https://github.com/dotnet/sdk) repository for further development and respond to issues formerly in this repository.
>
>All other analyzers moved into the [dotnet/roslyn](https://github.com/dotnet/roslyn) repository.

**Comment by SeSamSeg** [comment]
Looking at sdk project, It's a duplicate of https://github.com/dotnet/sdk/issues/43574 so I'll add my findings there.
